app/code/Magento/MediaGalleryMetadata/Model/Png/Segment/WriteXmp.php

```diff
@@ -97,15 +97,19 @@ public function execute(FileInterface $file, MetadataInterface $metadata): FileI
     }
 
     /**
-     * Insert XMP segment to image png segments (at position 1)
+     * Insert XMP segment to image png segments before IEND chunk
      *
      * @param SegmentInterface[] $segments
      * @param SegmentInterface $xmpSegment
      * @return SegmentInterface[]
      */
     private function insertPngXmpSegment(array $segments, SegmentInterface $xmpSegment): array
     {
-        return array_merge(array_slice($segments, 0, 2), [$xmpSegment], array_slice($segments, 2));
+        return array_merge(
+            array_slice($segments, 0, count($segments) - 1),
```

Comment:
I think extracting the variable `$iendSegmentIndex = count($segments) - 1` would not only avoid double `count` call but also positively affect the code readability

Reply:
ok